content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx

```diff
@@ -0,0 +1,186 @@
+---
+title: 'Names in the underlying database'
+metaDescription: ''
+tocDepth: 2
+---
+
+<TopBlock>
+
+The [Prisma schema](./) includes mechanisms that allow you to define names of certain database objects. You can:
+
+- [Map model and field names to different collection/table and field/column names in the underlying database](#mapping-collectiontable-and-fieldcolumn-names)
+- [Define constraint and index names](#constraint-and-index-names-preview) (Preview)
+
+</TopBlock>
+
+## Mapping collection/table and field/column names
+
+Prisma model [naming conventions (singular form, PascalCase)](../../../reference/api-reference/prisma-schema-reference#naming-conventions) <span class="api"></span> do not always match table names in the database. A common approach for naming tables/collections in databases is to use plural form and [snake_case](https://en.wikipedia.org/wiki/Snake_case) notation.
+
+For example, when you introspect a database with a table named `comments`, the result Prisma model will look like this:
+
+```prisma
+model comments {
+   // Fields
+}
+```
+
+However, you can still follow the naming convention without renaming the underlying `comments` table in the database by using the [`@@map`](../../../reference/api-reference/prisma-schema-reference#map-1) <span class="api"></span> attribute:
+
+```prisma
+model Comment {
+   // Fields
+  @@map("comments")
+}
+```
+
+With this modified model definition, Prisma automatically maps the `Comment` model to the `comments` table in the underlying database.
+
+You can also [`@map`](../../../reference/api-reference/prisma-schema-reference#map) <span class="api"></span> a column/field name:
+
+```prisma highlight=2-4;normal
+model Comment {
+  content String  @map("comment_text")
+  email   String  @map("commenter_email")
+  type    Enum    @map("comment_type")
+
+  @@map("comments")
+}
+```
+
+You can also `@map` an enum value, or `@@map` an enum:
+
+```prisma highlight=3,5;normal
+enum Type {
+  Blog,
+  Twitter @map("comment_twitter")
+
+  @@map("comment_source_enum")
+}
+```
+
+`@map` and `@@map` allow you to [tune the shape of your Prisma Client API](../prisma-client/working-with-prismaclient/use-custom-model-and-field-names#using-map-and-map-to-rename-fields-and-models-in-the-prisma-client-api) by decoupling model and field names from table and column names in the underlying database.
+
+## Constraint and index names (Preview)
+
+In [2.29.0](https://github.com/prisma/prisma/releases/tag/2.29.0) and later, you can use the `map` field to define the **underlying constraint and index names** in the Prisma schema. The `map` field is _only_ visible in the schema if you use a constraint name that differs from Prisma's naming convention.
+
```

Comment:
This sentence should be separate in my opinion and could also probably be refined a bit:

```suggestion

The `map` field is _only_ visible in the schema if you use a constraint name that differs from Prisma's naming convention.
```

Reply:
Yeah, the wording around map I also found difficult. Tricky to get right I think. 

Afaik `@map` and `@@map` we call attributes. Their specific properties within braces (for those two just `name`) are referred to as fields in the docs. So in the sentence above it would be the `map` field and not the `map` attribute. 

So you have `map` attributes that have a `name` field. And on the other hand you have attributes like `@@unique` that take a `map` and a `name` field that both hold some kind of name -.-

Reply:
I think the things of the "Attributes" are actually called "Arguments": https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#arguments-1

Reply:
Yeah, arguments is what they're also called in the Rust code. I also think this is cleaner